### ISSUE TITLE
Break out internal roles in claw-playbook

### ISSUE THREAD

**Issue by dannylamb** [Islandora-CLAW/CLAW#743]
In order to get them on galaxy and use them in more complicated multi-server setups, we need to break out our internal roles into their own github repositories in the Islandora-Devops organization.

Current list of roles:
 - [x] [activemq](https://github.com/Islandora-Devops/ansible-role-activemq)
 - [x] [alpaca](https://github.com/Islandora-Devops/ansible-role-alpaca)
 - [x] [apix](https://github.com/Islandora-Devops/ansible-role-apix)
 - [x] [blazegraph](https://github.com/Islandora-Devops/ansible-role-blazegraph)
 - [x] [cantaloupe](https://github.com/Islandora-Devops/ansible-role-cantaloupe)
 - [x] [crayfish](https://github.com/Islandora-Devops/ansible-role-crayfish)
 - [x] [fcrepo-syn](https://github.com/Islandora-Devops/ansible-role-fcrepo-syn)
 - [x] [fcrepo](https://github.com/Islandora-Devops/ansible-role-fcrepo)
 - [x] [fits](https://github.com/Islandora-Devops/ansible-role-fits)
 - [x] [grok](https://github.com/Islandora-Devops/ansible-role-grok)
 - [x] [karaf](https://github.com/Islandora-Devops/ansible-role-karaf)
 - [x] [openseadragon](https://github.com/Islandora-Devops/ansible-role-drupal-openseadragon)
 - [x] [tomcat8](https://github.com/Islandora-Devops/ansible-role-tomcat8)

**Comment by whikloj** [comment]
@dannylamb You want that I should make repos in the Islandora-DevOps organization for each of these? Is there a naming structure to follow?

**Comment by dannylamb** [comment]
Yeah, that'd be great.  Riffing from claw-playbook I guess we're hyphenating instead of underscoring?  I checked galaxy and there didn't seem to be a discernible convention.

**Comment by whikloj** [comment]
Okay as a note, to find out what platforms are available you can use this API
https://galaxy.ansible.com/api/v1/platforms/?format=api&name=Ubuntu
Not well documented, but there is a Github issue open that lead to it.

Also what versions of each platform do we want to support and test with? For instance, the above link list 16 versions for Ubuntu.

**Comment by whikloj** [comment]
Also tags..."Islandora" or "CLAW" or "Islandora-CLAW"?

**Comment by whikloj** [comment]
Actually tags are a single alpha-numeric term, so for Fedora I'm going with 
```
tags:
   - fedora
   - islandora
   - CLAW
   - repository
   - ldp
   - installer
```

**Comment by dannylamb** [comment]
@whikloj We're targeting Ubuntu xenial and Centos 7.  I'm assuming this is for advertising what they work on?

**Comment by whikloj** [comment]
Pardon that is `EL` not `Enterprise_Linux`

**Comment by whikloj** [comment]
All split out however I'm wondering if this is an [ActiveMQ](https://galaxy.ansible.com/shelleg/activemq/) role that we could put some effort behind. 

Has some downloads, but is missing xenial support. We could fork it and if it works provide some PRs. 

I have made the a repo for activemq, so if they are non-responsive we won't have lost this work.

Thoughts?

Also, I like one of the conventions out there which is providing variables for major, minor and patch version number.

Lastly, now I am going to start with some clean up as I found some variable bleed between roles. Perhaps I'll see if I can figure out how to write some tests and use that molecule thing @DiegoPino  pointed out.

**Comment by jonathangreen** [comment]
Should probably make a PR to remove the broken out roles from claw-playbook, and to add them to requirements.yml so we pull in our own roles from galaxy. I guess making a release for each of them is probably a good idea too. That maybe could be another ticket, just wanted to add my 2c.

**Comment by whikloj** [comment]
@jonathangreen Don't they need to be in galaxy before we can pull them in? Just because I think @dannylamb will need to create the islandora user account to get the nice `islandora.role-name` names.

**Comment by dannylamb** [comment]
@whikloj They may just have tags wrong in ansible-galaxy.  According to the repo's README it supports 16.04?  Maybe we should just give it a try.

I'll go make an account for islandora.

**Comment by whikloj** [comment]
[molecule](http://molecule.readthedocs.io/en/latest/) while interesting doesn't work out of the box on my Mac. It could be my docker setup. But I'm not sure.

I was following [this post](https://www.jeffgeerling.com/blog/testing-ansible-roles-travis-ci-github) which led me to start [this branch](https://github.com/whikloj/ansible-role-alpaca/tree/travis-tests) which is a good start (if I do say so myself).

However we will need to consider docker and something along the lines of (but hopefully way simpler than) this 
https://www.jeffgeerling.com/blog/2016/how-i-test-ansible-configuration-on-7-different-oses-docker

**Comment by whikloj** [comment]
https://travis-ci.org/whikloj/ansible-role-alpaca

**Comment by jonathangreen** [comment]
Whenever we are pulling the roles out, we need to make sure we sync them up first. There are probably other changes, but I wanted to make sure these two PR don't get lost:
https://github.com/Islandora-Devops/claw-playbook/pull/31
https://github.com/Islandora-Devops/claw-playbook/pull/32

**Comment by jonathangreen** [comment]
I think we can call this closed!
